index.js

```diff
@@ -1,47 +1,234 @@
-const request = require('request');
+var request = require('request');
 
-const slackWebhooks = (settings) => {
+var slackWebhooks = function() {
 
-	const slackWebhookUrl = settings.slackWebhookUrl;
-	const name = settings.name;
+	var LOG_LEVEL = 1;
+	var WARN_LEVEL = 2;
+	var ERROR_LEVEL = 3;
+	var ALL_LEVEL = 4;
+
+	var defaultUrl;
+	var name;
+	var channels;
 
 	/**
-	 * Overrides console.error to send error messages to Slack via webhooks.
+	 * Initializes console-to-slack to send messages to Slack on console usage.
 	 *
-	 * @function publicOverrideConsoleError
+	 * @function init
+	 *
+	 * @since 0.0.2
+	 * @access public
+	 * @memberOf console-to-slack
+	 * @param {string} url           The default url to send slack messages to.
+	 * @param {number} consoleLevel  Level which indicates which console usage to override.
+	 * @param {Object} options       Options configurable by the user.
+	 */
+	function init(url, consoleLevel, options) {
+
+		// Setup the default url
+		if (!url) {
+			throw new Error('You must provide a default url for console-to-slack to work correctly.');
+		}
+
+		defaultUrl = url;
+
+		// Setup the passed-in options
+		if (options && options.name) {
+			name = options.name;
+		}
+
+		if (options && options.channels) {
+			channels = options.channels;
+		}
+
+		// Setup the overrides depending on the chosen level
+		if (consoleLevel === LOG_LEVEL) {
+			_overrideLog();
+		} else if (consoleLevel === WARN_LEVEL) {
+			_overrideWarn();
+		} else if (consoleLevel === ERROR_LEVEL) {
+			_overrideError();
+		} else if (consoleLevel === ALL_LEVEL) {
+			_overrideLog();
+			_overrideWarn();
+			_overrideError();
+		} else {
+			throw new Error('You must provide a console level for console-to-slack to work correctly (1 = console.log, 2 = console.warn, 3 = console.error, 4 = all).');
+		}
+
+	}
+
+	/**
+	 * Overrides console.log to send the log messages to Slack via webhooks.
+	 * Note: Original console.log behaves as normal
+	 *
+	 * @function _overrideLog
+	 *
+	 * @since 0.0.2
+	 * @access public
```

Comment:
Thanks to your effective use of the [Revealing Module Pattern](https://addyosmani.com/resources/essentialjsdesignpatterns/book/#revealingmodulepatternjavascript), these would be private now!